scripts/ubuntu-bartender/azure-provider

```diff
@@ -0,0 +1,65 @@
+function _wait-instance-connectable() {
+  for attempt in $(seq 12); do
+    if ! ssh -qo StrictHostKeyChecking=no "ubuntu@$ip_addr" -- true; then
```

Comment:
how can I set the private key when using `AZURE_PUBKEY_PATH` ? the aws-provider adds some `-i ...` and just assumes that there might be a local file . Maybe I'm missing something here?

Reply:
Yes, I remember now and I overlooked this part.

Essentially, it is not possible on Azure to do the same as on AWS. Azure (from what I know) doesn't let you store your keys and re-use them on VMs[1].

[1] https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-ssh-keys-detailed

Reply:
You can now [store SSH keys in Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/ssh-keys-portal) in the portal.  It's not fully integrated into their instructions as your link doesn't show this code path, but it works.

Reply:
Nice to see you here @rcj4747 , I pushed an alternative solution that I prefer. But I can also revert it an use this feature if needed.